modules/base/tool.go

```diff
@@ -72,6 +76,21 @@ func DetectEncoding(content []byte) (string, error) {
 	return result.Charset, err
 }
 
+// DetectEncodingAndConvert detects the encoding of content and coverts to UTF-8 if possible
+func DetectEncodingAndConvert(content []byte) []byte {
```

Comment:
We already have the same function on `templates.ToUTF8WithErr(buf)`

Reply:
Yes but there are two problems with it. First it does return string this way I would have unneeded casting to string and back to byte array and second it does return error but I need to return default byte array in case of error